File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    temp_dict = {}
    
    for i in clothes:
        key, value = i[1],i[0]
        if key in temp_dict:
            temp_dict[key].append(value)
        else:
            temp_dict[key] = [value]

    for key in temp_dict.keys():
        if len(temp_dict) == 1 :
            return len(temp_dict[key])
        else:
            answer *= len(temp_dict[key]) + 1
    
    return answer-1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String,Integer> map = new HashMap<>();
        for(String[] items: clothes){
            map.put(items[1], map.getOrDefault(items[1],0)+1);
        }
        
        
        for (int count : map.values()) {
            answer *= (count + 1); // 선택하지 않는 경우 포함
        }
        
        return answer-1;
    }
}